tsdb/cursor.go

```diff
@@ -1,35 +1,22 @@
 package tsdb
 
 import (
-	"bytes"
 	"container/heap"
-)
-
-// Direction represents a cursor navigation direction.
-type Direction bool
+	"encoding/binary"
+	"sort"
+	"strings"
 
-const (
-	// Forward indicates that a cursor will move forward over its values.
-	Forward Direction = true
-	// Reverse indicates that a cursor will move backwards over its values.
-	Reverse Direction = false
+	"github.com/influxdb/influxdb/influxql"
 )
 
-func (d Direction) String() string {
-	if d.Forward() {
-		return "forward"
-	}
-	return "reverse"
-}
-
-// Forward returns true if direction is forward
-func (d Direction) Forward() bool {
-	return d == Forward
-}
+// EOF represents a "not found" key returned by a Cursor.
+const EOF = int64(-1)
 
-// Forward returns true if direction is reverse
-func (d Direction) Reverse() bool {
-	return d == Reverse
+// Cursor represents an iterator over a series.
+type Cursor interface {
+	Seek(seek int64) (key int64, value interface{})
```

Comment:
minor: if we really don't want to collied with vet (and I'm not that torn about it) this could be `Find` maybe? or perhaps `SeekKey`.  The vet warning we had to disable isn't one that I really care a lot about as it doesn't prevent bugs, it's more of a style warning.